netsync/manager.go

```diff
@@ -481,7 +482,7 @@ func (sm *SyncManager) handleCheckSyncPeer() {
 	}
 
 	best := sm.chain.BestSnapshot()
-	if topBlock == best.Height {
+	if topBlock == best.Height || sm.chain.UtxoCacheFlushInProgress() {
```

Comment:
It looks like the flush state write is protected by `utxoCache.mtx` but this read is not acquiring a lock/read lock at least up to the sm.blockHandler() level.

Reply:
yeah, do we think it needs to be protected here? If it changes shortly before this line is executed it doesn't seem like it materially affects the outcome does it?

Reply:
I'm still wrapping my head around the code so I can't answer confidently.

- If a flush is finished, but we detect that it is in progress, then no problem. We catch it later.

- If a flush is just starting but we detect it is not in progress, then it depends on the interaction with `peerStates[]`, `clearRequestedState()`, and `updateSyncPeer()`. Do you want me to investigate that? You probably know off hand whether it's important or not.

- If we locked it, then we may not even need the test since this code would be unreachable during a flush.

Reply:
Yeah I'm not sure if it will hurt anything. If it thinks a flush is not in progress when it is then it might increment the counters but the next time through the loop it should detect it properly if the flush hasn't finished yet.

Reply:
I think we are ok here.